packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts

```diff
@@ -175,6 +176,45 @@ export async function injectRecaptchaFields<T>(
   return newRequest;
 }
 
+type ActionMethod<TRequest, TResponse> = (
+  auth: Auth,
+  request: TRequest
+) => Promise<TResponse>;
+
+export async function handleRecaptchaFlow<TRequest, TResponse>(
+  authInstance: AuthInternal,
+  request: TRequest,
+  actionName: RecaptchaActionName,
+  actionMethod: ActionMethod<TRequest, TResponse>
+): Promise<TResponse> {
+  if (authInstance._getRecaptchaConfig()?.emailPasswordEnabled) {
```

Comment:
Can we also include "provider" name or enum as a parameter? Based on that, we will know whether to check emailPasswordEnabled or a different provider. 

Ok to handle this the next time we add a provider too. Please add a TODO in that case.

Reply:
Yeah, as an internal method, I prefer add the parameter only when needed so the context is more clear.